### DISCUSSION TITLE
Best way to update to latest Version?

### DISCUSSION THREAD

**Discussion by seanbts** [sinamics/ztnet#208] [Q&A]
Hi,

Currently running V3.11 on standalone debian install, what's the best way to update ztnet to latest v4 version?

Thanks

**Comment by sinamics** [comment]
latest version should be installed by just run the installation command again.
https://ztnet.network/installation/linux

**Reply by seanbts** [reply]
Thanks, all done and working

**Comment by Urajmal** [comment]
Will add this to the topic: 

I have a problem of updating to 0.5.3. I guess it is fault in my RaspberryPI 3B. I am using Raspbian, with ZTnet version 0.4.0. I have installed btop, to monitor the system status, and at some point on the second picture always freezes. I was thinking that at first there is problem with temperature, so I cooled the RPi down to 40°C but still it stops responding at random file when installing npm. The result is the web interface does not show up, and sais connection refused. However the Zerotier One is running, if I check at my connected devices. Any suggestion, how can I overcome this? I am using a Linux version, not Docker.

![kitty_portable-0 76 1 12_RzsngIv8jQ](https://github.com/sinamics/ztnet/assets/81016616/85789154-dd55-4744-a6b7-3e586b6795cf)
![kitty_portable-0 76 1 12_jANiSc2JHS](https://github.com/sinamics/ztnet/assets/81016616/3ec3defd-23e1-417b-b5c9-655b20a10568)

Thank you!

**Reply by Urajmal** [reply]
OK So I manage to get further with addressing a lot more Swap, but I get stopped again at picture bellow. 
![kitty_portable-0 76 1 12_boJLeBwQaQ](https://github.com/sinamics/ztnet/assets/81016616/9cd9bd55-844e-48e9-b9f4-f4d84cae2c70)

**Comment by sinamics** [comment]
Thank you for reporting.
I will test myself on a Raspberry 3B, but i guess 1GB RAM is at the limit of what is required to build the application.

**Reply by Urajmal** [reply]
Yes, I saw that if I make a 2MB of swap it goes over the first point of installing nmp, but it stops at "Checking validity of types ..."

The zerotier-one service is running, I am able to connect, so the base exist, but is there any way to run web gui somehow (if we account in the update up to the point above?) Meaning, what is missed after the "Checking validity of types ..."?

Thank you!

**Reply by sinamics** [reply]
Your best bet is to run ztnet with docker as the ztnet application is pre-built inside the container.

**Reply by Urajmal** [reply]
Thank you for your answer! 

If you could a little explain (if you have time), how can I run Ztnet in Docker on Debian linux, I will be very happy? I am not into Docker for now, but I see I need to study the topic. I am also unaware how can Docker application use PostgreSQL and node which are not in Docker.

Thanks for the help!

**Reply by sinamics** [reply]
Try this, from the top of my head:
install docker:
```bash
curl -fsSL https://get.docker.com -o get-docker.sh
sh get-docker.sh
```

then create a new file somewhere, example /home/docker-compose.yml
and paste the content from the docs
https://ztnet.network/installation/docker-compose#setup

Then run `docker compose pull`, this will download all the images, and then finally run `docker compose up -d` to start all services.
